platform/ios/src/MGLMapboxEvents.m

```diff
@@ -456,6 +456,8 @@ - (void) pushTurnstileEvent {
         
         // Flush
         [strongSelf flush];
```

Comment:
Agreed @friedbunny. For this PR, I just wanted to make sure that the turnstile event post continues to always be attempted, expediently, irrespective of the state of telemetry. I think that larger refactors to separate out networking and to remove threading complexity may make things clearer here, too. It might make sense to have a different queue or no queue at all for the turnstile event (i.e. just post it right away).